LFS-1145: Modify set_admin_password.py so that it can also change passwords for non-admin accounts

IntegralProgrammer opened the pull request:
This PR modifies `set_admin_password.py` and gives it the ability to set the password for non-admin Sling users.

To test:
1. Build this branch (`mvn clean install`)
2. Start CARDS
3. Create a new CARDS user named `test` with a password of `12345678`
4. `python3 Utilities/Administration/set_admin_password.py --user test`
5. Enter `test1234` for the `test` user password
6. `python3 Utilities/Administration/set_admin_password.py`
7. Enter `admin5678` for the `admin` user password
8. You should now be able to log in to CARDS as the `test` user with the password `test1234` and as the `admin` user with the password `admin5678`

IntegralProgrammer committed:
LFS-1145: Modify set_admin_password.py so that it can also change passwords for non-admin accounts

sdumitriu commented on Utilities/Administration/set_admin_password.py:
```diff
@@ -22,9 +22,14 @@
 
```
I think the script file should be renamed as well, since it no longer sets just the admin password.

IntegralProgrammer replied:
Sure, I could rename it to `set_sling_password.py` as LFS-930 has not yet been merged and renaming this script won't break anything.

IntegralProgrammer replied:
Implemented by commit `29b833d` :heavy_check_mark:

sdumitriu commented on Utilities/Administration/set_admin_password.py:
```diff
@@ -22,9 +22,14 @@
 import requests
 from requests.auth import HTTPBasicAuth
 
+argparser = argparse.ArgumentParser()
```
I don't like inconsistencies, and we use envvars to specify the URL and the admin password, but only a command line argument for the user. Ideally, all 3 parameters should support arg, with fallback on envvar, with fallback on default value.

IntegralProgrammer replied:
It would be a bad practice to pass the _admin password_ as a command line argument as it could be observed by anyone who is able to run `ps aux`, as well, it would be logged in the bash history. So the _admin password_ should always be passed as an environment variable with fallback on a default value.

As for the _URL_ argument, this is not considered sensitive and therefore can be passed as command line argument with fallback to an environment variable with fallback to a default value.

As for the _user_ argument, it is currently read as a command line argument with fallback to a default value. I am hesitant to support reading it from an environment variable as unlike the _URL_ and _admin password_ variables which would highly likely remain constant between calls to `set_admin_password.py` the _user_ argument could change and falling back to a previously set environment variable may be unexpected behavior for the user.

So I propose that I just change the way _URL_ is handled, first check command line arguments, then fallback to an environment variable, then fallback to a default value. How does that sound?

sdumitriu replied:
Sounds good.

IntegralProgrammer replied:
Implemented by commit `37bda6e` :heavy_check_mark:

IntegralProgrammer committed:
LFS-1145: Modify set_admin_password.py so that it can also change passwords for non-admin accounts

For obtaining the URL for the running CARDS instance, first try the command line arguments, then fallback to the CARDS_URL environment variable, and then finally fallback to the constant http://localhost:8080

IntegralProgrammer committed:
LFS-1145: Modify set_admin_password.py so that it can also change passwords for non-admin accounts

Rename set_admin_password.py to set_sling_password.py as that utility program now covers non-admin Sling users

sdumitriu commented on Utilities/Administration/set_sling_password.py:
```diff
@@ -22,25 +22,35 @@
   ADMIN_PASSWORD = os.environ["ADMIN_PASSWORD"]
 
-new_password = input("New admin password: ")
+new_password = input("New {} password: ".format(args.user))
```
```suggestion
new_password = input("New password for {}: ".format(args.user))
```

sdumitriu commented on Utilities/Administration/set_sling_password.py:
```diff
@@ -22,25 +22,35 @@
+change_req = requests.post(pw_change_url, data=form_data, auth=HTTPBasicAuth('admin', ADMIN_PASSWORD))
 if change_req.status_code != 200:
-  print("Error while setting admin password")
+  print("Error while setting {} password".format(args.user))
```
```suggestion
  print("Error while setting password for {}".format(args.user))
```

sdumitriu commented on Utilities/Administration/set_sling_password.py:
```diff
@@ -22,25 +22,35 @@
+  print("Error while setting {} password".format(args.user))
   sys.exit(-1)
-print("Set admin password")
+print("Set {} password".format(args.user))
```
```suggestion
print("Password for {} has been updated".format(args.user))
```

sdumitriu committed:
[misc] Improve wording
